### PR TITLE
add migration to migrate redis for new parties with a contribution

### PR DESCRIPTION
**Pull request by marcusmolchany**
* add `/migrations/*` dir to hold redis migrations
* roughly following Rails Active Record format https://guides.rubyonrails.org/active_record_migrations.html

### CODE REVIEW

File: src/fetchers.ts [hunk 1]
```diff
@@ -176,3 +176,42 @@ export const getLastKnownBlockNumber = async (): Promise<number> => {
...
+  // return zr.party_contribution.map(
+  //   (c): ContributionPartyEvent => {
+  //     return {
+  //       eventType: "contribution",
+  //       party: c.party,
+  //       txHash: c.transactionHash,
+  //       contribution: {
+  //         contributorAddress: c.contributedBy,
+  //         amountInEth: (parseInt(c.contributedAmountWei) / 10 ** 18).toString(),
+  //         totalAmountContributedToPartyInWei: BigNumber.from(
```
Comment (steveklebanoff) [inline src/fetchers.ts]: we technically dont really need all these fields from here, right? just one?

File: src/migrations/20220218021400_update_party_first_contribution.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * forward migration - set redis to "true" for all parties that have a contribution from
+ * someone that is not the party creator
+ */
+export const up = async () => {
+  console.log("starting up migration - set new party with contribution");
+  const partyAddressWithContributions = await getAllPartyAddressesWithContributions();
+
+  for (let i = 0; i < partyAddressWithContributions.length; i++) {
+    const partyAddress = partyAddressWithContributions[i];
+    console.info(`setting new party with contribution ${partyAddress} to true`);
```
Comment (steveklebanoff) [inline src/migrations/20220218021400_update_party_first_contribution.ts:16]: are the party addresses always expected to be lowercase?

**Review by steveklebanoff — CHANGES_REQUESTED** [review]

File: src/migrations/20220218021400_update_party_first_contribution.ts (hunk 2 above)
Comment (marcusmolchany) [reply src/migrations/20220218021400_update_party_first_contribution.ts:16]: oh good call

Comment (marcusmolchany) [reply src/migrations/20220218021400_update_party_first_contribution.ts:16]: we're always using the values from hasura when we get a `partyAddress`, and afaik those party addresses are always lowercased. i can explicitly lowercase these values.

**Pull request merged**